src/aws/osml/model_runner/app.py

```diff
@@ -758,10 +758,69 @@ def aggregate_features(
         if image_request_item.tile_size and image_request_item.tile_overlap:
             # Read all the features from DDB.
             features = feature_table.get_features(image_request_item.image_id)
+            logger.info(f"Total features aggregated: {len(features)}")
         else:
             raise AggregateFeaturesException("Tile size and overlap must be provided for feature aggregation")
         return features
 
+    @staticmethod
+    def identify_overlap(
+        feature: Feature, shape: Tuple[int, int], overlap: Tuple[int, int], origin: Tuple[int, int] = (0, 0)
+    ) -> Tuple[int, int, int, int]:
+        """
+        Generate a tuple that contains the min and max indexes of adjacent tiles or regions for a given feature. If
+        the min and max values for both x and y are the same then this feature does not touch an overlap region.
+
+        :param feature: the geojson Feature that must contain properties to identify its location in an image
+        :param shape: the width, height of the area in pixels
+        :param overlap: the x, y overlap between areas in pixels
+        :param origin: the x, y coordinate of the area in relation to the full image
+
+        :return: a tuple: minx, maxx, miny, maxy that identifies any overlap.
+        """
+        bbox = feature["properties"][GeojsonDetectionField.BOUNDS]
+
+        # If an offset origin was supplied adjust the bbox so the key is relative to the origin.
+        bbox = (bbox[0] - origin[0], bbox[1] - origin[1], bbox[2] - origin[0], bbox[3] - origin[1])
+
+        stride_x = shape[0] - overlap[0]
+        stride_y = shape[0] - overlap[1]
+
+        max_x_index = int(bbox[2] / stride_x)
+        max_y_index = int(bbox[3] / stride_y)
+
+        min_x_index = int(bbox[0] / stride_x)
+        min_y_index = int(bbox[1] / stride_y)
+        min_x_offset = int(bbox[0]) % stride_x
+        min_y_offset = int(bbox[1]) % stride_y
+
+        if min_x_offset < overlap[0] and min_x_index > 0:
+            min_x_index -= 1
+        if min_y_offset < overlap[1] and min_y_index > 0:
+            min_y_index -= 1
+
+        return min_x_index, max_x_index, min_y_index, max_y_index
+
+    @staticmethod
+    def group_features_by_overlap(
```

Comment:
Unit test?

Reply:
Will follow up w/ a tasks to move this logic and implement specific unit testing.